docs/src/io-service.md

```diff
@@ -0,0 +1,11 @@
+---
+short-description: The IO service.
+...
+
+# IO service
+
+The IO service is an user-provided class which manages waiting for events and waking up coroutines/operations based on them.
+
+The IO service must provide one method: `void wait()`. This method is called when there is no more work to do currently. It waits for any event to happen, and wakes up the appropriate coroutine/operation which awaited the event.
```

Comment:
maybe a basic flow chart or even a small event loop (or a link to one) as an illustration of how they work?

Reply:
I've added a textual description for this. As discussed this will be turned into a block diagram later.